### PR TITLE
Books api

### PR DESCRIPTION
**Pull request by alejoaquili**
## Summary
List of books integration with API REST using redux.

#### Nexus 5X
Peek 2019-04-23 10-44

## Trello Card
https://trello.com/c/AXI88Ptt/26-n-102-integracion-con-books

### CODE REVIEW

File: src/services/BookService.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import api from '../config/api';
+
+
+const ENDPOINT = '/books';
+
+export default {
+    getBooks: () => api.get(ENDPOINT)
```
Comment (guidoprinc) [inline src/services/BookService.js]: BOOKS_ENDPOINT

File: src/app/screens/BookList/index.js [hunk 2]
```diff
@@ -1,34 +1,41 @@
...
+
+    handlePressBook = (item) => {
+        const {navigation} = this.props;
+        navigation.navigate(BookDetail, {book: item});
+    }
+
+    async componentDidMount() {
+        const { getBooks } = this.props;
+        await getBooks();
+    }
```
Comment (guidoprinc) [inline src/app/screens/BookList/index.js]: do you really need this `async/await`?

File: src/app/screens/BookList/layout.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+class BookList extends PureComponent {
+
+  renderItem = ({item}) =>  <Book name={item.title} author={item.author} imageSource={{uri: `https${item.image.url.substring(4)}?id=${item.id}`}} onPress={() => this.props.handlePressBook(item)}/>;
+
+  keyExtractor = (item) => `${item.id}`;
+
+  render() {
+    const { books, getBooksErrorMessage } = this.props;
+
+    const errorMessage = !getBooksErrorMessage ?  'No books available at library.' : getBooksErrorMessage;
```
Comment (sfernandez11) [inline src/app/screens/BookList/layout.js]: Move `'No books available at library.'` into a constant

File: src/app/screens/BookDetail/index.js [hunk 4]
```diff
@@ -4,14 +4,15 @@ import { View } from 'react-native';
...
+import { defaultComments } from './comments';
 
 class BookDetail extends PureComponent {
     render() {
       const book = this.props.navigation.getParam('book');
       return (
         <View style={styles.container}>
-          <BookDetailComponent title= {book.title} author={book.author} year={book.year} type={book.type} imageSource={book.imageSource}/>
-          <CommentList comments={book.comments}></CommentList>
+          <BookDetailComponent title= {book.title} author={book.author} year={book.year} type={book.genre} imageSource={{uri: `https${book.image.url.substring(4)}?id=${book.id}`}}/>
```
Comment (sfernandez11) [inline src/app/screens/BookDetail/index.js]: Create a function to do this
```
`https${book.image.url.substring(4)}?id=${book.id}`
```
And use it here

File: src/app/screens/BookList/layout.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import { FlatList, Text } from 'react-native';
+
+import Book from './components/Book';
+import styles from './styles';
+
+import WithLoading from '../../components/WithLoading';
+
+class BookList extends PureComponent {
+
+  renderItem = ({item}) =>  <Book name={item.title} author={item.author} imageSource={{uri: `https${item.image.url.substring(4)}?id=${item.id}`}} onPress={() => this.props.handlePressBook(item)}/>;
```
Comment (sfernandez11) [inline src/app/screens/BookList/layout.js]: And here! Also destruct `this.props`

File: src/app/screens/BookList/layout.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    const errorMessage = !getBooksErrorMessage ?  'No books available at library.' : getBooksErrorMessage;
+    const error = <Text style={styles.errorMessage}>{errorMessage}</Text>;
+    
+    return (
+      <FlatList 
+        data={books} 
+        renderItem={this.renderItem} 
+        keyExtractor={this.keyExtractor} 
+        contentContainerStyle={styles.container}
+        ListEmptyComponent={error}
```
Comment (sfernandez11) [inline src/app/screens/BookList/layout.js]: Put directly the component here: `ListEmptyComponent={<Text style={styles.errorMessage}>{getBooksErrorMessage ? getBooksErrorMessage : NO_AVAILABLE_BOOKS}</Text>}`

File: src/app/screens/BookList/layout.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import React, { PureComponent } from 'react';
+import { FlatList, Text } from 'react-native';
+
+import Book from './components/Book';
+import styles from './styles';
+
+import WithLoading from '../../components/WithLoading';
+import { NO_BOOKS } from '../../../constants/errors';
+
+export const generateBookUri = book => `https${book.image.url.substring(4)}?id=${book.id}`
```
Comment (sfernandez11) [inline src/app/screens/BookList/layout.js]: ⚠️ Do you think this is the right place to put this function declaration? Maybe in an utiles file?

Comment (sfernandez11) [inline src/app/screens/BookList/layout.js]: Do you think this is the right place to put this function? You are using this function on two different screens, maybe you can move it to a `bookUtils` file inside a `utils` folder in the `src/`. Also, you can return an object like
```
export const generateBookUri = book => ({ uri: `https${book.image.url.substring(4)}?id=${book.id}` });
```
And then: `imageSource={generateBookUri(item)}`

**Pull request merged**